lib/krane/deploy_task.rb

```diff
@@ -238,18 +210,22 @@ def secrets_from_ejson
     def discover_resources
       @logger.info("Discovering resources:")
       resources = []
-      crds_by_kind = cluster_resource_discoverer.crds.group_by(&:kind)
+      crds_grouped = cluster_resource_discoverer.crds.group_by(&:cr_group_kind)
+      group_kinds = @task_config.group_kinds
+
       @template_sets.with_resource_definitions(current_sha: @current_sha, bindings: @bindings) do |r_def|
-        crd = crds_by_kind[r_def["kind"]]&.first
+        group = ::Krane::KubernetesResource.group_from_api_version(r_def["apiVersion"])
+
+        crd = crds_grouped[::Krane::KubernetesResource.combine_group_kind(group, r_def["kind"])]&.first
         r = KubernetesResource.build(namespace: @namespace, context: @context, logger: @logger, definition: r_def,
-          statsd_tags: @namespace_tags, crd: crd, global_names: @task_config.global_kinds)
+          statsd_tags: @namespace_tags, crd: crd, group_kinds: group_kinds)
```

Comment:
Same comment as in global_deploy_task about using cluster_resource_discoverer here instead of passing the list of group kinds to `build`.